### DISCUSSION TITLE
Modal dialog root

### DISCUSSION THREAD

**Discussion by Mudloop** [fomantic/Fomantic-UI#1343] [Modules (Modal, Dropdown, Popup, Calendar, Toast, ...)] [answered]
Hi,

Part of my page can be made fullscreen by my users. The problem with this is that any modal popups are not visible when they are requested, because those live in the root of the document (they do show up on the page, but are not visible when another element is fullscreen).

I've tried reparenting the modal dialogs so they are children of the fullscreen element, but this doesn't seem to work. Looks like they get reparented by the code. If I place a modal inside another element, I'm doing so for a good reason, and it should stay there (and place the dimmer inside that element.

Is there a way to set a modal's root element? And if not, could someone please point me in the right direction as to how I could achieve this?

Thanks!

**Comment by lubber-de** [comment, marked as answer]
Basically modals are supposed to be defined as direct child of the body. However you can define your modals html code inside other child nodes, as you said, and try if `detachable:false` works in your case then. You might also try to change the `context` setting to your own modals desired root node (default is body). 
To better understand your particular case and probably find a solution for you, would you be able to provide a jsfiddle to better understand the whole situation?

**Comment by Mudloop** [comment]
@lubber-de Thank you for your help! Changing the context setting did the trick.
